### ISSUE TITLE
Issues with greedy_facility_location.py 

### ISSUE THREAD

**Issue by pavanteja295** [ozansener/active_learning_coreset#3]
Hi,
  I think the file greedy_facility_location.py  is not executable because I can see in the 
line 31
`#for lab in dat['gt_f'].shape[0]):` There are some unbalanced parenthesis

line 38 
`no = numpy.argmax(d)`  . d is not defined.

It would be great if you can share your code which you used for the results obtained in the ICLR paper

**Comment by pavanteja295** [comment]
Also according to the paper you should be using Subset s for calculating the distance matrix if I'm not wrong. But in this code you are not computing the distance matrix with the subset . This file is also not error free . Can you please fix the errors?

**Comment by pavanteja295** [comment]
Most of the code is actually not complete.
The implementation details provided in the paper are different from those that in the repo. Can you please let us know why?

**Comment by ozansener** [comment]
This is only the active learning part. The actual training part is not here. Unfortunately, we can not open source it but it is based on http://torch.ch/blog/2015/07/30/cifar.html. 

If you only need the greedy solver, you can use this active learning module: https://github.com/google/active-learning/blob/master/sampling_methods/kcenter_greedy.py which also implements our method.

Full solver is functional if you generate and pickle the embeddings. I will add a how-to document for it.

**Comment by pavanteja295** [comment]
Can you please let me know the implementation details of the method. Like
learning rate epochs per each iteration.. Etc etc

**Comment by ozansener** [comment]
We trained for 200 Epochs and half the LR at every 50 epochs. We are using random horizontal flips. We subtracted average R,G,B values from respective channels. And, whitened images before training.

**Comment by pavanteja295** [comment]
Cool thanks a lot Ozan. But it was mentioned that lr was constant with
0.0001 and rms prop so it's a bit conflicting.
   Also these settings are valid for cifar-100 also I hope. Most
importantly for reducing the weights of the input neurons, alpha in the
paper you are using regularizer I suppose?

**Comment by ozansener** [comment]
What is alpha? I do not remember any alpha in the paper. 

Cifar 10 and 100 uses the same script. 

As I mentioned, the network is exactly same as here: http://torch.ch/blog/2015/07/30/cifar.html We only used its Tensorflow version. They use SGD, wee use RMSprop thats the only difference. You can use that script and save the feature embeddings to file and use our code for active learning.

**Comment by pavanteja295** [comment]
Sorry my bad.

Also the ratio of labelled images mentioned in the graphs refer to total
number of labelled images ?

Ratio of labelled images = [0.1, 0.2, ...1] So are you training with 5k
samples when you mean 0.1 for which you get an accuracy of ~30% ?

**Comment by ozansener** [comment]
Yes, the ratio is to the full training set.

**Comment by redsadaf** [comment]
> This is only the active learning part. The actual training part is not here. Unfortunately, we can not open source it but it is based on http://torch.ch/blog/2015/07/30/cifar.html.
> 
> If you only need the greedy solver, you can use this active learning module: https://github.com/google/active-learning/blob/master/sampling_methods/kcenter_greedy.py which also implements our method.
> 
> Full solver is functional if you generate and pickle the embeddings. I will add a how-to document for it.

Hi,
Do you plan to add a "how_to" document anytime soon?!? It would be a great help to understand your method.

Thanks

**Comment by ozansener** [comment]
I will make an update soon probably in the month of November with the detailed readme and training details.

**Comment by redsadaf** [comment]
> I will make an update soon probably in the month of November with the detailed readme and training details.
Thanks Ozan,
I ran your code, and final results are some models saved as "*.sol" format( such as "s_100_solution_0.055231944963340884.sol"). Now how could I find which points have been chosen to be sent to the oracle?

**Comment by ozansener** [comment]
@redsadaf 

You can use the script at https://github.com/ozansener/active_learning_coreset/blob/master/coreset/gurobi_solution_parser.py 
to parse the results

**Comment by fatemehtd** [comment]
Hi Ozan,
Do you still have a plan to make an update soon with the detailed readme and training details?

**Comment by linlinlin1993** [comment]
Hi Ozan,
Could you please update the details of training settings and how to use your code?

**Comment by ozansener** [comment]
Please use the code from: https://github.com/google/active-learning/blob/master/sampling_methods/kcenter_greedy.py We are not supporting this anymore.
